resolwe/flow/managers/listener.py

```diff
@@ -0,0 +1,253 @@
+from resolwe.permissions.utils import copy_permissions
+from resolwe.utils import BraceMessage as __
+
+if settings.USE_TZ:
+    from django.utils.timezone import now  # pylint: disable=ungrouped-imports
+else:
+    import datetime
+    now = datetime.datetime.now  # pylint: disable=invalid-name
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class ExecutorListener(Thread):
+    """Main listener implementation."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize attributes."""
+        super(ExecutorListener, self).__init__(*args, **kwargs)
+        self._redis = redis.StrictRedis(**kwargs)
+        self._terminated = Event()
+
+        self._main_db = db.connections[db.DEFAULT_DB_ALIAS]
+        self._saved_sharing = self._main_db.allow_thread_sharing
+
+        if 'verbosity' in kwargs:
+            self._verbosity = kwargs['verbosity']
+        else:
+            self._verbosity = 1
+
+    def __enter__(self):
+        """On entering a `with` statement, start the thread and return this object."""
+        self.start()
+        return self
+
+    def __exit__(self, typ, value, trace):
+        """On exiting a `with` statement, kill the listener and wait for it.
+
+        Exceptions are all propagated.
+        """
+        self.terminate()
+        self.join()
+
+        # re-raise all exceptions
+        return False
+
+    def hydrate_spawned_files(self, exported_files_mapper, filename, data_id):
+        """Hydrate spawned files' paths."""
+        if filename not in exported_files_mapper[data_id]:
+            raise KeyError('Use `re-export` to prepare the file for spawned process: {}'.format(filename))
+
+        export_fn = exported_files_mapper[data_id].pop(filename)
+
+        if exported_files_mapper[data_id] == {}:
+            exported_files_mapper.pop(data_id)
+
+        return {'file_temp': export_fn, 'file': filename}
+
+    def start(self, *args, **kwargs):
+        """Start the thread and allow database connection sharing."""
+        self._main_db.allow_thread_sharing = True
+        super(ExecutorListener, self).start(*args, **kwargs)
+
+    def join(self, *args, **kwargs):
+        """Wait for the thread to finish and reset connection sharing back to original settings."""
+        super(ExecutorListener, self).join(*args, **kwargs)
+        self._main_db.allow_thread_sharing = self._saved_sharing
+
+    def terminate(self):
+        """Stop the standalone manager."""
+        self._terminated.set()
+
+    @staticmethod
+    def channel_name_pair():
+        """Return the names of the executor result channel name (first), and the feedback channel name (second)."""
+        flow_dict = getattr(settings, 'FLOW_EXECUTOR', {})
+        return (
+            flow_dict.get('RESULT_CHANNEL', 'resolwe_manager.result_queue'),
+            flow_dict.get('RESULT_FEEDBACK_CHANNEL_PREFIX', 'resolwe_manager.executor_feedback'),
+        )
+
+    def _reply_chan(self, obj):
```

Comment:
Rename to `_reply_channel`.

Reply:
Renamed to `_queue_response_channel` for consistency with other places.